torch/distributed/fsdp/_fully_shard/_fsdp_param.py

```diff
@@ -796,6 +797,19 @@ def shard_mesh(self):
             return mesh[mesh.mesh_dim_names[-1]]
         raise ValueError(f"Invalid mesh: {mesh}")
 
+    @property
+    def shard_mesh_from_root(self):
+        mesh = self.mesh_info.mesh
+
+        if mesh.ndim == 1:
+            return mesh
+        else:
+            assert mesh.mesh_dim_names is not None
```

Comment:
`assert mesh.mesh_dim_names is not None` is quite strong. wonder if see such assumption somewhere in FSDP2? I was thinking of fixing this from device mesh api

Reply:
@weifengpy I think for HSDP path `mesh.ndim == 2`, we do assume that mesh dim names are specified.

Reply:
From this PR: https://github.com/pytorch/pytorch/pull/137436

Reply:
got you. the main question is whether it's acceptable to use `_MeshEnv.get_root_mesh`. We should have @wz337 tomorrow for discussion

Reply:
I think it should be acceptable but I could be wrong